src/service/Offers.ts

```diff
@@ -41,33 +40,47 @@ export const getOfferService = (
       id,
       instanceHandle,
       publicInvitationMaker,
-      proposalTemplate: { give, want },
+      proposalTemplate: { give: giveTemplate, want: wantTemplate },
     } = offer;
 
-    const mapPursePetnamesToBrands = paymentProposals =>
-      Object.fromEntries(
+    const convertProposals = async paymentProposals => {
+      const entries = await Promise.all(
         Object.entries(paymentProposals).map(
           // @ts-expect-error
-          ([kw, { pursePetname, value }]) => {
-            const brand = pursePetnameToBrand.get(pursePetname);
-            if (!brand) {
+          async ([kw, { pursePetname, value, amount: serializedAmount }]) => {
+            if (!serializedAmount && !pursePetnameToBrand.get(pursePetname)) {
               return [];
             }
-            return [
-              kw,
-              {
-                brand,
-                value: BigInt(value),
-              },
-            ];
+
+            /// XXX test e2e with dapp inter once feasible.
+            const amount = serializedAmount
+              ? await E(boardIdMarshaller).unserialize(serializedAmount)
+              : AmountMath.make(
+                  pursePetnameToBrand.get(pursePetname),
+                  BigInt(value),
+                );
+
+            return [kw, amount];
           },
         ),
       );
+      return Object.fromEntries(entries);
+    };
 
-    const instance = await E(boardIdMarshaller).unserialize(instanceHandle);
-    const {
-      slots: [instanceBoardId],
-    } = await E(boardIdMarshaller).serialize(instance);
+    const deconstructInstance = async () => {
+      const instance = await E(boardIdMarshaller).unserialize(instanceHandle);
+      const {
+        slots: [instanceBoardId],
+      } = await E(boardIdMarshaller).serialize(instance);
```

Comment:
i had to remember there's no promise pipelining in this app so the two awaits are fine.

it wasn't obvious to me why you wanted to wrap these in a function. It looks like you want `convertProposals` to resolve in parallel to resolving the `instance`. Good idea. But I don't see why you want `instanceBoardId` to be available after this Promise.all.

consider,
```
    const [instance, give, want] = await Promise.all([
      E(boardIdMarshaller).unserialize(instanceHandle),
      convertProposals(giveTemplate),
      convertProposals(wantTemplate),
    ]);
```
and then just above the first use of `instanceBoardId`,
```
    const {
      slots: [instanceBoardId],
    } = await E(boardIdMarshaller).serialize(instance);
```

Reply:
Done.

I'm not sure about promise pipelining in the rest of the app, but we await here so that the card doesn't show up before it has the data needed to render.